assets/javascripts/modules/autoCompleteFactory.js

```diff
@@ -2,15 +2,37 @@ require('jquery');
 var autoComplete = require('./autoComplete.js');
 
 var createAutoCompleteCountries = function () {
-  var $chooseCountryAutoCompleteElem = $('.js-choose-country-auto-complete');
-  var $countryCodeInput = $('#countryCode');
-  var suggestionDisplayTemplate = function (title, value) {
-    return title + ' (+' + value + ')';
-  };
+  $('.js-choose-country-auto-complete').each(function(){
+    
+    var $countryCodeInput;
+    if($('#countryCode').length > 0){
+      $countryCodeInput = $('#countryCode');
+    } else {
+      $countryCodeInput = $('[id="' + $(this).attr("data-suggestions") + '"]');    
+    }
```

Comment:
so `data-suggestions` is used for validation [here](https://github.com/hmrc/assets-frontend/blob/master/assets/javascripts/validation/customValidations.js#L12) and the `$countryCodeInput` is used as the `$targetInputElem` described [here](https://github.com/hmrc/assets-frontend/blob/master/assets/javascripts/modules/autoComplete.js#L325)

So with those explanations (`data-suggestions` validation needs documenting) I think you would agree it feels a bit mixed up whats going on here. 

What you are wanting to say is target your select. This "target" element should be sent in via a new `data-`. 

For backwards compatibility you need to keep the `$countryCodeInput` work (as you have done) but I would move this check inside the `autoComplete.js` file [here](https://github.com/hmrc/assets-frontend/blob/master/assets/javascripts/modules/autoComplete.js#L333) Then you can check to see if it is an element or a string and do the appropriate.